src/main/java/com/snowflake/kafka/connector/Utils.java

```diff
@@ -454,16 +473,29 @@ && parseTopicToTableMap(config.get(SnowflakeSinkConnectorConfig.TOPICS_TABLES_MA
               DELIVERY_GUARANTEE,
               SnowflakeSinkConnectorConfig.IngestionDeliveryGuarantee.AT_LEAST_ONCE.name()));
     } catch (IllegalArgumentException exception) {
-      LOGGER.error(
-          "Delivery Guarantee config:{} error:{}", DELIVERY_GUARANTEE, exception.getMessage());
-      configIsValid = false;
+      invalidConfigParams.put(
+          DELIVERY_GUARANTEE,
+          Utils.formatString(
+              "Delivery Guarantee config:{} error:{}", DELIVERY_GUARANTEE, exception.getMessage()));
     }
 
     // Check all config values for ingestion method == IngestionMethodConfig.SNOWPIPE_STREAMING
-    final boolean isStreamingConfigValid = StreamingUtils.isStreamingSnowpipeConfigValid(config);
+    invalidConfigParams.putAll(StreamingUtils.validateStreamingSnowpipeConfig(config));
+
+    // log all invalid params and throw exception
+    if (!invalidConfigParams.isEmpty()) {
+      String invalidParamsMessage = "";
+
+      for (String invalidKey : invalidConfigParams.keySet()) {
+        String invalidValue = invalidConfigParams.get(invalidKey);
+        String errorMessage =
+            Utils.formatString(
+                "Config value '{}' is invalid. Error message: '{}'", invalidKey, invalidValue);
+        invalidParamsMessage += errorMessage + "\n";
+      }
 
-    if (!configIsValid || !isStreamingConfigValid) {
-      throw SnowflakeErrors.ERROR_0001.getException();
+      LOGGER.error("Invalid config: " + invalidParamsMessage);
```

Comment:
Here is where we both log the invalid params and throw an exception. Potentially not necessary to do both, however I wanted to retain the previous behavior (logs invalid param and also changes boolean to false)